pkg/cableengine/syncer/syncer.go

```diff
@@ -53,7 +54,7 @@ func (i *GatewaySyncer) syncGatewayStatus() {
 
 	gatewayObj, err := i.generateGatewayObject()
 	if err != nil {
-		klog.Errorf("Error generating Gateway object: %s", err)
+		utilruntime.HandleError(fmt.Errorf("Error generating Gateway object: %s", err))
```

Comment:
Nit:

In our case HandleError doesn't buy much:
   * checks for nil (we already need to do that for return)
   * does error backoff if errors happen <1ms , but this function is time gated.

Reply:
Using `HandleError` gives the unit tests a mechanism to verify something when testing error paths. Also from the docs "_Invoking this method is preferable to logging the error - the default behavior is to log but the errors may be sent to a remote server for analysis."_ .  This is used in other places in the submariner code base as well by the original author(s).